### PR TITLE
[libc] enable most of the entrypoints on aarch64

### PR DESCRIPTION
**Pull request by SchrodingerZhu**
This is a non-feature change that enables most of the entrypoints for aarch64 based runtime builds. It fixes an additional problem that some compiler-rt targets are not defined at the time of dependency checking thus leading to false-negatives.

### CODE REVIEW

File: libc/config/linux/aarch64/entrypoints.txt [hunk 1]
```diff
@@ -552,17 +575,25 @@ if(LIBC_TYPES_HAS_FLOAT16)
     libc.src.math.canonicalizef16
     libc.src.math.ceilf16
     libc.src.math.copysignf16
+    libc.src.math.expf16
```
Comment (overmighty) [inline libc/config/linux/aarch64/entrypoints.txt:578]: We ran into an issue with `expf16` on the AArch64 buildbots that I couldn't reproduce on my AArch64 phone: https://github.com/llvm/llvm-project/pull/100632#issuecomment-2258772681.

File: libc/config/linux/aarch64/entrypoints.txt [hunk 2]
```diff
@@ -552,17 +575,25 @@ if(LIBC_TYPES_HAS_FLOAT16)
...
     libc.src.math.f16add
     libc.src.math.f16addf
+    libc.src.math.f16addl
     libc.src.math.f16div
     libc.src.math.f16divf
+    libc.src.math.f16divl
+    libc.src.math.f16fma
```
Comment (overmighty) [inline libc/config/linux/aarch64/entrypoints.txt:585]: I think `f16fma` was unnecessarily disabled (disabling `f16fmal` and `f16fmaf128` should have been enough), but all other disabled `float16` entrypoints except `expf16` caused internal errors in Clang 11, so unless we've bumped our minimum supported Clang version and updated Clang on the Buildbot workers, we can't enable these entrypoints yet.

File: libc/config/linux/aarch64/entrypoints.txt [hunk 3]
```diff
@@ -552,17 +575,27 @@ if(LIBC_TYPES_HAS_FLOAT16)
...
+    # TODO: aarch64 bug
+    # Please see https://github.com/llvm/llvm-project/pull/100632#issuecomment-2258772681
+    # libc.src.math.expf16
     libc.src.math.f16add
     libc.src.math.f16addf
+    # libc.src.math.f16addl
     libc.src.math.f16div
     libc.src.math.f16divf
+    # libc.src.math.f16divl
+    # libc.src.math.f16fma
```
Comment (overmighty) [inline libc/config/linux/aarch64/entrypoints.txt:587]: I think you can enable `f16fma`. We lost the Buildbot logs, but we still have a GitHub comment from me saying that the issue was ISel failing with `fp_round` from `f128` to `f16`: https://github.com/llvm/llvm-project/pull/97072#discussion_r1658925698.

I added `f16fma` in the same PR as `f16fmal` and `f16fmaf128`, that's why I disabled all three of them. Then I realized that `double` shouldn't cause issues, so I only enabled `float16` narrowing functions that take `float` or `double` on AArch64 in following PRs, and it didn't cause issues.

**Review by overmighty — APPROVED** [review]

**Comment by SchrodingerZhu** [comment]
not failing on my machine.

**Pull request merged**